src/main/java/ch/hevs/cloudio/endpoint/JobsManager.java

```diff
@@ -0,0 +1,141 @@
+
+    public static JobsManager getInstance(){
+        return instance;
+    }
+
+    public List<String> listScripts(String filePath){
+        File folder = new File(filePath);
+        File[] listOfFiles = folder.listFiles();
+        ArrayList result = new ArrayList();
+
+        for (File listOfFile : listOfFiles) {
+            if (listOfFile.isFile()) {
+                result.add(listOfFile.getName());
+            }
+        }
+        return result;
+    }
+
+    public void executeJob(String jobURI, String filePath, Boolean output, MqttAsyncClient mqtt, String uuid){
+        ProcessBuilder processBuilder = new ProcessBuilder();
+
+        String scheme = jobURI.split("://")[0];
+        String job = jobURI.split("://")[1];
+
+        switch (scheme) {
+            case "file":
+
+                if (!listScripts(filePath).isEmpty() && listScripts(filePath).contains(job)){
+                    processBuilder.command("bash", "-c", "sh " + filePath + "/" + job);
+
+                    try {
+
+                        Process process = processBuilder.start();
+
+                        if (!output)
+                            return;
+
+                        BufferedReader reader =
+                                new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+                        String line;
+                        while ((line = reader.readLine()) != null) {
+                            mqtt.publish("@execOutput/" + uuid, line.getBytes(), 1, false);
+                        }
+
+                        int exitCode = process.waitFor();
+                        System.out.println("\nExited with error code : " + exitCode);
+
+                    } catch (MqttException exception) {
+                        log.error("Exception: " + exception.getMessage());
+                        exception.printStackTrace();
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }else
+                {
+                    try {
+                        mqtt.publish("@execOutput/" + uuid, ("file uri:" + jobURI + " contain invalid file").getBytes(), 1, false);
+                    } catch (MqttException exception) {
+                        log.error("Exception: " + exception.getMessage());
+                        exception.printStackTrace();
+                    }
+                }
+                break;
+            case "cmd":
+                try {
+                    System.out.println(job);
+                    CmdJobs cmdJobs = CmdJobs.valueOf(job);
+                    switch (cmdJobs){
+                        case listJobs:
+                            List<String> files = JobsManager.getInstance().listScripts(filePath);
+                            try {
+                                mqtt.publish("@execOutput/" + uuid, "List of jobs available as jobURI \"cmd://commandName\":".getBytes(), 1, false);
```

Comment:
Would it not be simpler to publish a message containing all output and newlines rather than sending multiple messages?

Reply:
Since I send line by line the output of a script execution, I stayed with the same idea here where one line is sent in one message

Reply:
Ok, but then we might need to add a correlation ID to the command and each line of the output contains that ID too, otherwise there might be problems to isolate the output of different commands in history and in the case of concurrent command executions this can cause some confusion.

Reply:
I like this idea, it will go nice also with the rest version of jobs that use "Server-sent events" that can contain id in the answer